tests/test_dvc.py

```diff
@@ -68,15 +68,22 @@ def __call__(self, exp_ref):
 @pytest.mark.parametrize("save", [True, False])
 def test_exp_save_on_end(tmp_dir, mocker, save):
     dvc_repo = mocker.MagicMock()
+    dvc_repo.index.stages = []
     with mocker.patch("dvclive.live.get_dvc_repo", return_value=dvc_repo):
         live = Live(save_dvc_exp=save)
         live.end()
     if save:
+        assert live._baseline_rev is not None
+        assert live._exp_name is not None
         dvc_repo.experiments.save.assert_called_with(
             name=live._exp_name, include_untracked=live.dir
         )
+        assert (tmp_dir / live.dvc_file).exists()
```

Comment:
@dberenbaum Your call on what to do with make_dvcyaml in a follow-up.
Added the test cases here